### PR TITLE
Documment logging formats

### PR DESCRIPTION
**Pull request by serathius**
During recent WG Structured Logging meeting we went into discussion what features/changes should be make in specific problem. This lead us to agree that we need some high level guide about what changes make sense in which format. 

This is draft PR that I started writing to document goals of each logging format.
/cc @ehashman @pohly

### CODE REVIEW

File: contributors/devel/sig-instrumentation/logging.md [hunk 1]
```diff
@@ -39,7 +39,18 @@ As per the comments, the practical default level is V(2). Developers and QE
 environments may wish to run at V(3) or V(4). If you wish to change the log
 level, you can pass in `-v=X` where X is the desired maximum level to log.
 
-## Logging header formats
+## Logging Formats
+
+Kubernetes supports multiple logging formats to allow users flexibility in picking format that best matches their needs.
```
Comment (pohly) [inline contributors/devel/sig-instrumentation/logging.md:44]:
```suggestion
Kubernetes supports multiple logging formats to allow users flexibility in picking a format that best matches their needs.
```

File: contributors/devel/sig-instrumentation/logging.md [hunk 2]
```diff
@@ -39,7 +39,18 @@ As per the comments, the practical default level is V(2). Developers and QE
...
+As new feature and logging formats are proposed we expect that Kubernetes community might find itself in disagreement about what formats and features should be officially supported.
```
Comment (pohly) [inline contributors/devel/sig-instrumentation/logging.md:45]:
```suggestion
As new features and logging formats are proposed, we expect that the Kubernetes community might find itself in disagreement about what formats and features should be officially supported.
```

File: contributors/devel/sig-instrumentation/logging.md [hunk 3]
```diff
@@ -39,7 +39,18 @@ As per the comments, the practical default level is V(2). Developers and QE
...
+This section will serve as a developer documentation describing high level goals of each format, giving a holistic vision of logging options and allow community to resolve any future disagreement when making changes.
```
Comment (pohly) [inline contributors/devel/sig-instrumentation/logging.md:46]:
```suggestion
This section will serve as a developer documentation describing high level goals of each format, giving a holistic vision of logging options and allowing the community to resolve any future disagreement when making changes.
```

File: contributors/devel/sig-instrumentation/logging.md [hunk 4]
```diff
@@ -39,7 +39,18 @@ As per the comments, the practical default level is V(2). Developers and QE
...
+
+### Text logging format
+
+* Default format in Kubernetes.
+* Purpose: Maintain backward compatibility with klog format.
+* Human-readable, multiline support.
```
Comment (pohly) [inline contributors/devel/sig-instrumentation/logging.md:52]: I think human-readable is also a purpose, not just some accidental aspect of it. I would write it like this:
```
* Purpose:
  * Maintain backward compatibility with klog format.
  * Human-readable, with support for printing multi-line strings with actual line breaks.
```

File: contributors/devel/sig-instrumentation/logging.md [hunk 5]
```diff
@@ -59,4 +70,12 @@ where the fields are defined as follows:
...
 	msg              The user-supplied message
 ```
+
 See more in [here](https://github.com/kubernetes/klog/blob/9ad246211af1ed84621ee94a26fcce0038b69cd1/klog.go#L581-L597)
+
+### JSON logging format
+
+* Requires passing `--logging-format=json` to enable
+* Purpose: Provide structured metadata in efficient way.
+* Machine-readable, no multiline support.
```
Comment (pohly) [inline contributors/devel/sig-instrumentation/logging.md:80]:
```suggestion
* Machine-readable, multi-line strings are quoted to fit into a single output line.
```

File: contributors/devel/sig-instrumentation/logging.md [hunk 6]
```diff
@@ -59,4 +70,12 @@ where the fields are defined as follows:
...
+* Full support for structured metadata.
```
Comment (pohly) [inline contributors/devel/sig-instrumentation/logging.md:81]:
```suggestion
* Full support for structured metadata, using `logr.MarshalLog` when available.
```

**Review by pohly — CHANGES_REQUESTED** [review]

**Comment by pohly** [comment]
Looks good, thanks for capturing this.

File: contributors/devel/sig-instrumentation/logging.md (hunk 4 above)
Comment (serathius) [reply contributors/devel/sig-instrumentation/logging.md:52]: I will fix it in git. For future https://github.blog/changelog/2020-02-26-multi-line-code-suggestions-beta/

**Comment by serathius** [comment]
@pohly addressed all comments.

**Review by pohly — APPROVED** [review]
/lgtm

**Pull request merged**